### Eclipse Workspace Patch 1.0
#P Core
Index: src/java/org/jpox/store/rdbms/table/ClassTable.java
===================================================================
RCS file: /cvsroot/jpox/JPOX/Core/src/java/org/jpox/store/rdbms/table/ClassTable.java,v
retrieving revision 1.225
diff -u -r1.225 ClassTable.java
--- src/java/org/jpox/store/rdbms/table/ClassTable.java	23 May 2007 15:02:13 -0000	1.225
+++ src/java/org/jpox/store/rdbms/table/ClassTable.java	24 May 2007 15:21:37 -0000
@@ -2020,14 +2020,16 @@
         Iterator cks = candidateKeysByMapField.values().iterator();
         int ckNum = 0;
 
+        PrimaryKey pk = this.getPrimaryKey();
         RDBMSIdentifierFactory idFactory = (RDBMSIdentifierFactory)storeMgr.getIdentifierFactory();
         while (cks.hasNext())
         {
             DatastoreIdentifier ckName = idFactory.newCandidateKeyIdentifier(this, ++ckNum);
             CandidateKey ck = (CandidateKey)cks.next();
-            ck.setName(ckName.getIdentifier());
-
-            stmts.add(dba.getAddCandidateKeyStatement(ck, idFactory));
+            if(!pk.getColumnList().equals(ck.getColumnList())) {
+            	ck.setName(ckName.getIdentifier());
+            	stmts.add(dba.getAddCandidateKeyStatement(ck, idFactory));
+            }
         }
 
         return stmts;
