### Eclipse Workspace Patch 1.0
#P Core
Index: src/java/org/jpox/store/rdbms/adapter/PostgreSQLAdapter.java
===================================================================
RCS file: /cvsroot/jpox/JPOX/Core/src/java/org/jpox/store/rdbms/adapter/PostgreSQLAdapter.java,v
retrieving revision 1.48
diff -u -r1.48 PostgreSQLAdapter.java
--- src/java/org/jpox/store/rdbms/adapter/PostgreSQLAdapter.java	3 May 2007 14:04:46 -0000	1.48
+++ src/java/org/jpox/store/rdbms/adapter/PostgreSQLAdapter.java	15 May 2007 12:17:44 -0000
@@ -179,6 +179,7 @@
             psqlTypes.put("" + Types.DOUBLE,"float8");
             psqlTypes.put("" + Types.INTEGER,"int4");
             psqlTypes.put("" + Types.LONGVARCHAR,"text");
+            psqlTypes.put("" + Types.CLOB,"text");
             psqlTypes.put("" + Types.NUMERIC,"numeric");
             psqlTypes.put("" + Types.REAL,"float4");
             psqlTypes.put("" + Types.SMALLINT,"int2");
