### Eclipse Workspace Patch 1.0
#P Core
Index: src/java/org/jpox/store/StoreManager.java
===================================================================
RCS file: /cvsroot/jpox/JPOX/Core/src/java/org/jpox/store/StoreManager.java,v
retrieving revision 1.151
diff -u -r1.151 StoreManager.java
--- src/java/org/jpox/store/StoreManager.java	23 May 2007 20:50:18 -0000	1.151
+++ src/java/org/jpox/store/StoreManager.java	24 May 2007 15:09:40 -0000
@@ -563,7 +563,7 @@
      */
     public void addClass(String className, ClassLoaderResolver clr)
     {
-        addClasses(new String[] {className}, clr, null, false);
+        addClasses(new String[] {className}, clr, null, null, false);
     }
     
     /**
@@ -576,13 +576,13 @@
      * runtime to do it on-demand.
      * @param classes The class(es) to be added.
      * @param clr The ClassLoaderResolver
-     * @param writer Optional writer when you just want the DDL for persisting the specified classes
+     * @param ddlWriter Optional writer when you just want the DDL for persisting the specified classes
      * @param completeDdl whether complete DDL will be created when writing DDL to a file, or only for missing elements
      * @exception org.jpox.store.exceptions.DatastoreValidationException
      *      If there is some mismatch between the current datastore contents and
      *      those necessary to enable persistence of the given classes.
      */
-    public abstract void addClasses(String[] classes, ClassLoaderResolver clr, Writer writer, boolean completeDdl);
+    public abstract void addClasses(String[] classes, ClassLoaderResolver clr, Writer ddlWriter, Writer sqlWriter, boolean completeDdl);
 
    /**
      * Add classes to the persistence model for the datastore.
@@ -600,7 +600,7 @@
      */
      public void addClasses(String[] classNames, ClassLoaderResolver clr)
      {
-         addClasses(classNames, clr, null, false);
+         addClasses(classNames, clr, null, null, false);
      }
 
     /**
